### PR TITLE
Fix kontrast failing for documents with just comments

### PR DESCRIPTION
**Pull request by jaseemabid**

### CODE REVIEW

File: pkg/k8s/resource_helper.go [hunk 1]
```diff
@@ -102,6 +104,11 @@ func (rh *ResourceHelper) NewResourceFromBytes(bytes []byte) (*Resource, error)
 	// format, API group, kind, version
 	obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(bytes, nil, nil)
 
+	// A missing `Kind` most probably meant an empty document
+	if runtime.IsMissingKind(err) {
```
Comment (milesbxf) [inline pkg/k8s/resource_helper.go]: I wonder if we could more explicitly exclude empty documents? I think we'd still want to pick up a valid use case where someone has accidentally missed off the `Kind`

Comment (jaseemabid) [reply pkg/k8s/resource_helper.go]: @milesbxf Decode function in `scheme.Codecs.UniversalDeserializer().Decode(bytes, nil, nil)` decodes the yaml and immediately looks for a `Kind` before doing anything else, there doesn't seem to be a more specific error than this one unfortunately.  

https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/helpers.go#L326-L344

Comment (milesbxf) [reply pkg/k8s/resource_helper.go]: Urgh, that's annoying. Perhaps we could directly decode the YAML ourselves, exclude the empty documents, then pass to the decoder as bytes?

Comment (jaseemabid) [reply pkg/k8s/resource_helper.go]: @milesbxf We could do a `gvk.Empty()` as well, will that work? Found this. 

```patch
- obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(bytes, nil, nil)
+ obj, gvk, err := scheme.Codecs.UniversalDeserializer().Decode(bytes, nil, nil)
```

```
// Empty returns true if group, version, and kind are empty
func (gvk GroupVersionKind) Empty() bool {
	return len(gvk.Group) == 0 && len(gvk.Version) == 0 && len(gvk.Kind) == 0
}
```

Comment (milesbxf) [reply pkg/k8s/resource_helper.go]: Hmm - is `gvk` a pointer? If not, `gvk` will be empty whenever there is any kind of error

**Review by milesbxf — APPROVED** [review]
Nice - this looks great 👌

File: pkg/k8s/resource_helper.go [hunk 2]
```diff
@@ -73,9 +74,20 @@ func (rh *ResourceHelper) NewResourcesFromFilename(filename string) ([]*Resource
...
+		}
+
+		// Converting to JSON and looking for "null" ignores empty docs.
+		bs, err = yaml.ToJSON(bs)
+
+		if err != nil {
+			return []*Resource{}, fmt.Errorf("failed to convert yaml to json %s: %s", filename, err.Error())
+		}
+
+		if len(bs) == 0 || bytes.Equal(bs, []byte("null")) {
```
Comment (milesbxf) [inline pkg/k8s/resource_helper.go:90]: I've just noticed a bug here - we return from the function and quit parsing the manifest if we get a `null` document, which means that any manifest which starts with an empty document will have all its resources skipped over

**Pull request merged**